Update GitHub checks api project page (#3661)
jenkins-infra/jenkins.io: content/projects/gsoc/2020/projects/github-checks.adoc

Thus, any information during the Jenkins process like warnings, summaries, and durations can be directly shown on GitHub pages.

=== Project Deliverables
* A new Jenkins plugin (link:https://github.com/jenkinsci/checks-api-plugin[Checks API Plugin]) that defines the general checks API

> * A new Jenkins plugin (link:https://github.com/jenkinsci/github-checks-plugin[GitHub Checks Plugin]) that implements the geneal checks API for GitHub

Reviewer: ```suggestion
* A new Jenkins plugin (link:https://github.com/jenkinsci/github-checks-plugin[GitHub Checks Plugin]) that implements the general checks API for GitHub
```

> * A new feature (link:https://github.com/jenkinsci/warnings-ng-plugin/pull/550[pull request #550]) for plugin:warnings-ng[Warnings Next Generation] plugin to report warnings to GitHub by consuming the checks API

Reviewer: this line and the next are inconsistent with plugin being in the link or not, should be consistent, no real opinion on which is best
```suggestion
* A new feature (link:https://github.com/jenkinsci/warnings-ng-plugin/pull/550[pull request #550]) for plugin:warnings-ng[Warnings Next Generation plugin] to report warnings to GitHub by consuming the checks API
```

* A new feature (link:https://github.com/jenkinsci/code-coverage-api-plugin/pull/169[pull request #169]) for plugin:code-coverage-api[Code Coverage API plugin] to report code coverage to GitHub by consuming the checks API

> * Pipeline support feature (link:https://github.com/jenkinsci/checks-api-plugin/pull/7[pull request #7]) to allow users publishing customized checks

Reviewer: ```suggestion
* Pipeline support feature (link:https://github.com/jenkinsci/checks-api-plugin/pull/7[pull request #7]) to allow users to publish checks from their pipeline
```

=== Project Deliverables
* A new Jenkins plugin (link:https://github.com/jenkinsci/checks-api-plugin[Checks API Plugin]) that defines the general checks API
* A new Jenkins plugin (link:https://github.com/jenkinsci/github-checks-plugin[GitHub Checks Plugin]) that implements the geneal checks API for GitHub
* A new feature (link:https://github.com/jenkinsci/warnings-ng-plugin/pull/550[pull request #550]) for plugin:warnings-ng[Warnings Next Generation] plugin to report warnings to GitHub by consuming the checks API
* A new feature (link:https://github.com/jenkinsci/code-coverage-api-plugin/pull/169[pull request #169]) for plugin:code-coverage-api[Code Coverage API plugin] to report code coverage to GitHub by consuming the checks API
* Pipeline support to allow users publishing customized checks
* Rerun support on GitHub to allow users requesting a rebuild for failed builds

=== Phase 1 Works
.Demo [starts from 50:15]
video::HQLhakpx5mk[youtube,width=800,height=420]

* link:https://github.com/jenkinsci/checks-api-plugin[Checks API Plugin]
* link:https://github.com/jenkinsci/github-checks-plugin[GitHub Checks Plugin]

> * link:/blog/2020/07/2020-07-09-github-checks-api-plugin-coding-phase-1.adoc[Blog post].

Author: I'll try to restore my slides for phase 1&2, since every time I just reused slides from the last phase and forget to store a copy...
